public/app/features/provisioning/SyncRepository.tsx

```diff
@@ -40,7 +40,10 @@ export function SyncRepository({ repository }: Props) {
     if (!name) {
       return;
     }
-    syncResource({ name, body: { complete: true } });
+    const elapsed = Date.now() - (repository.status?.sync.started ?? 0);
+    const complete = elapsed < 10000; // If clicked recently, do complete
+    console.log('Queue sync task', { elapsed, complete });
+    syncResource({ name, body: { complete } });
```

Comment:
should the sync button always be `complete`?  Only incremental when from a webhook?

This may be too funky, but it makes things incremental *unless* the sync was called within the last 10s